src/streamlink/plugins/svtplay.py

```diff
@@ -1,92 +1,146 @@
-        if match:
-            vid = match.group("id")
-            res = self.session.http.get(API_URL.format(vid))
-
-            videos = self.session.http.json(res, schema=_video_schema)
-            mapper = StreamMapper(cmp=lambda format, video: video["format"] == format)
-            mapper.map("hls", self._create_streams, "HLS", HLSStream.parse_variant_playlist)
-            mapper.map("hds", self._create_streams, "HDS", HDSStream.parse_manifest)
-        else:
-            res = self.session.http.get(self.url, params=dict(output="json"))
-            videos = self.session.http.json(res, schema=_old_video_schema)
+        path = self.url_re.match(self.url).group('path')
+        log.debug("Path={0}".format(path))
+
+        if path.startswith('/kanaler/'):
+            match = self.live_id_re.search(path)
+            if match is not None:
+                live_id = "ch-{0}".format(match.group('live_id'))
+                log.debug("Live ID={0}".format(live_id))
+            else:
+                raise PluginError("Failed to get live ID")
+
+            res = self.session.http.get(self.api_url.format(live_id))
+            api_data = self.session.http.json(res, schema=self._video_schema)
 
-            mapper = StreamMapper(cmp=lambda type, video: video["playerType"] == type)
-            mapper.map("ios", self._create_streams, "HLS", HLSStream.parse_variant_playlist)
-            mapper.map("flash", self._create_streams, "HDS", HDSStream.parse_manifest)
+            if 'programTitle' in api_data:
+                self.author = api_data['programTitle']
 
-        return mapper(videos)
+            self.category = 'Live'
+
+            if 'episodeTitle' in api_data:
+                self.title = api_data['episodeTitle']
+
+            for playlist in api_data['videoReferences']:
+                if playlist['format'] == 'dashhbbtv':
+                    for s in DASHStream.parse_manifest(self.session, playlist['url']).items():
+                        yield s
+        else:
+            res = self.session.http.get(self.url)
+            match = self.latest_episode_url_re.search(res.text)
+            if match:
+                res = self.session.http.get(
+                    urljoin(self.url, match.group('url')),
+                )
+
+            match = self.vod_id_re.search(res.text)
+
+            if match is not None:
+                vod_id = match.group('vod_id')
+                log.debug("VOD ID={0}".format(vod_id))
+            else:
+                raise PluginError("Failed to get VOD ID")
+
+            res = self.session.http.get(self.api_url.format(vod_id))
+            api_data = self.session.http.json(res, schema=self._video_schema)
+
+            if 'programTitle' in api_data:
+                self.author = api_data['programTitle']
+
+            self.category = 'VOD'
+
+            if 'episodeTitle' in api_data:
+                self.title = api_data['episodeTitle']
```

Comment:
The author, category and title logic could be made more DRY. Move this to a separate method, eg. `_set_metadata`.